results/cmd/api/main.go

```diff
@@ -19,28 +19,63 @@ package main
 import (
 	"context"
 	"database/sql"
+	"flag"
+	"fmt"
 	"log"
 	"net"
+	"os"
 
-	_ "github.com/mattn/go-sqlite3"
+	_ "github.com/go-sql-driver/mysql"
 	pb "github.com/tektoncd/experimental/results/proto/proto"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-const port = ":50051"
+const (
+	dbname     = "tekton-results"
+	dbprotocol = "tcp"
+)
+
+var dbaddr = flag.String("db_addr", "mysql.tekton-pipelines.svc.cluster.local", "Address of MySQL database to use.")
 
 func main() {
-	// Connect to sqlite DB.
-	db, err := sql.Open("sqlite3", "./results.db")
+	flag.Parse()
+
+	user, pass := os.Getenv("DB_USER"), os.Getenv("DB_PASSWORD")
+	if user == "" || pass == "" {
+		log.Fatal("Must provide both DB_USER and DB_PASSWORD")
+	}
+
+	if *dbaddr == "" {
+		log.Fatal("Must provide --db_addr")
+	}
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = ":50051"
```

Comment:
I suspect this was chosen because it's some sort of known default? Are there docs to point to?

Reply:
Honestly, I don't know if it's a default port. It seems to be commonly used in gRPC examples, which is where I picked it up ([e.g.](https://grpc.io/docs/guides/auth/#go)). It only really has to match whatever's in `config/api.yaml`.

Reply:
Let's add a comment to link to that file.

Reply:
Done.